cmake/flash/CMakeLists.txt

```
@@ -1,20 +1,64 @@
 # SPDX-License-Identifier: Apache-2.0
 
-add_custom_target(runner_yml_props_target)
-
-function(runner_yml_write content)
+function(runners_yaml_append content)
   # Append ${content}\n to a target property which is later evaluated as a
   # generator expression when writing the flash runner yaml file.
   # We define this function here to have access to the `flash` target.
 
   set_property(
-    TARGET         runner_yml_props_target
+    TARGET         runners_yaml_props_target
     APPEND_STRING
     PROPERTY       yaml_contents
     "${content}\n"
     )
 endfunction()
 
+function(get_runners_prop prop out_var)
+  # Get property 'prop' from runners_yaml_props_target. An optional
+  # third argument can be used to provide a default value if the value
+  # is not found (set to ...-NOTFOUND).
+
+  get_target_property(out runners_yaml_props_target "${prop}")
+
+  if(("${ARGC}" EQUAL 3) AND ("${out}" STREQUAL "out-NOTFOUND"))
+    set("${out_var}" "${ARGV2}" PARENT_SCOPE)
```

if making 3rd argument mandatory, then this can be reduced to:
```suggestion
  if("${out}" STREQUAL "out-NOTFOUND")
    set("${out_var}" "${default_value}" PARENT_SCOPE)
```